packages/legacy/core/App/contexts/auth.tsx

```diff
@@ -132,6 +135,28 @@ export const AuthProvider: React.FC = ({ children }) => {
     await wipeWalletKey(true)
   }
 
+  const rekeyWallet = async (oldPin: string, newPin: string, useBiometry?: boolean): Promise<boolean> => {
+    const secret = await loadWalletSalt()
+    if (!secret) {
+      return false
+    }
+    const oldHash = await hashPIN(oldPin, secret.salt)
+    const newSecret = await secretForPIN(newPin)
+    const newHash = await hashPIN(newPin, newSecret.salt)
```

Comment:
These `await` statements aren't wrapped in a try catch block in case something goes wrong with them, we might want to change that

Reply:
I don't think awaiting something has the chance to create an error does it? await just resolves a promise but there's no timeout or anything that would cause await to throw an error, right?